server/src/main/java/io/crate/rest/action/HttpErrorStatus.java

```diff
@@ -0,0 +1,90 @@
+    PARTITION_UNKNOWN(HttpResponseStatus.NOT_FOUND, 4046, "Unknown Partition"),
+    REPOSITORY_UNKNOWN(HttpResponseStatus.NOT_FOUND, 4047, "Unknown Repository"),
+    SNAPSHOT_UNKNOWN(HttpResponseStatus.NOT_FOUND, 4048, "Unknown Snapshot"),
+    USER_DEFINED_FUNCTION_UNKNOWN(HttpResponseStatus.NOT_FOUND, 4049, "Unknown user-defined function"),
+    USER_UNKNOWN(HttpResponseStatus.NOT_FOUND, 40410, "Unknown user"),
+    DOCUMENT_WITH_THE_SAME_PRIMARY_KEY_EXISTS_ALREADY(HttpResponseStatus.CONFLICT,4091, "A document with the same primary key exists already"),
+    VERSION_CONFLICT(HttpResponseStatus.CONFLICT,4092, "A VersionConflict. Might be thrown if an attempt was made to update the same document concurrently"),
+    RELATION_WITH_THE_SAME_NAME_EXISTS_ALREADY(HttpResponseStatus.CONFLICT,4093, "A relation with the same name exists already"),
+    TABLE_ALIAS_CONTAINS_TABLES_WITH_DIFFERENT_SCHEMA(HttpResponseStatus.CONFLICT,4094, "The used table alias contains tables with different schema"),
+    REPOSITORY_WITH_SAME_NAME_EXISTS_ALREADY(HttpResponseStatus.CONFLICT,4095, "A repository with the same name exists already"),
+    SNAPSHOT_WITH_SAME_NAME_EXISTS_ALREADY(HttpResponseStatus.CONFLICT,4096, "A snapshot with the same name already exists in the repository"),
+    PARTITION_FOR_THE_SAME_VALUE_EXISTS_ALREADY(HttpResponseStatus.CONFLICT,4097, "A partition for the same values already exists in this table"),
+    USER_DEFINED_FUNCTION_WITH_SAME_SIGNATURE_EXISTS_ALREADY(HttpResponseStatus.CONFLICT,4098, "A user-defined function with the same signature already exists"),
+    USER_WITH_SAME_NAME_EXISTS_ALREADY(HttpResponseStatus.CONFLICT,4099, "A user with the same name already exists"),
+    UNHANDLED_SERVER_ERROR(HttpResponseStatus.INTERNAL_SERVER_ERROR, 5000, "Unhandled server error"),
+    EXECUTION_OF_TASK_FAILED(HttpResponseStatus.INTERNAL_SERVER_ERROR,5001, "The execution of one or more tasks failed"),
+    ONE_OR_MORE_SHARDS_NOT_AVAILABLE(HttpResponseStatus.INTERNAL_SERVER_ERROR,5002, "One or more shards are not available"),
+    QUERY_FAILED_ON_ONE_OR_MORE_SHARDS(HttpResponseStatus.INTERNAL_SERVER_ERROR,5003, "The query failed on one or more shards"),
+    CREATING_SNAPSHOT_FAILED(HttpResponseStatus.INTERNAL_SERVER_ERROR,5004, "Creating a snapshot failed"),
+    QUERY_KILLED_BY_STATEMENT(HttpResponseStatus.INTERNAL_SERVER_ERROR, 5030, "The query was killed by a kill statement");
+
+    private final HttpResponseStatus httpResponseStatus;
+    private final int errorCode;
+    private final String message;
+
+    HttpErrorStatus(HttpResponseStatus httpResponseStatus, int erroCode, String message) {
+        this.httpResponseStatus = httpResponseStatus;
+        this.errorCode = erroCode;
+        this.message = message;
+    }
+
+    public HttpResponseStatus httpResponseStatus() {
+        return httpResponseStatus;
+    }
+
+    public int errorCode() {
+        return errorCode;
+    }
+
+    public String message() {
```

Comment:
Do we use the message from here? 

Looks like we have 2 sources of messages, once from the exception and once from the definition here. I think we should stick to the one from the exception and only add/annotate the response status and error code